exporter/collector/logs.go

```diff
@@ -283,41 +318,80 @@ func (l logMapper) logToEntry(
 
 	httpRequestAttr, ok := log.Attributes().Get(HTTPRequestAttributeKey)
 	if ok {
-		httpRequest, err := l.parseHTTPRequest(httpRequestAttr)
+		var httpRequest *logging.HTTPRequest
+		httpRequest, err = l.parseHTTPRequest(httpRequestAttr)
 		if err != nil {
 			l.obs.log.Debug("Unable to parse httpRequest", zap.Error(err))
 		}
 		entry.HTTPRequest = httpRequest
 	}
 
 	if log.SeverityNumber() < 0 || int(log.SeverityNumber()) > len(severityMapping)-1 {
-		return entry, fmt.Errorf("Unknown SeverityNumber %v", log.SeverityNumber())
+		return []logging.Entry{entry}, logName, fmt.Errorf("Unknown SeverityNumber %v", log.SeverityNumber())
 	}
 	entry.Severity = severityMapping[log.SeverityNumber()]
 
-	payload, err := parseEntryPayload(log.Body())
+	// Calculate the size of the internal log entry so this overhead can be accounted
+	// for when determining the need to split based on payload size
+	// TODO(damemi): Find an appropriate estimated buffer to account for the LogSplit struct as well
+	logOverhead, err := l.logEntryToInternal(entry, logName, mr, 0, 0)
 	if err != nil {
-		return entry, nil
+		return []logging.Entry{entry}, logName, err
 	}
-	entry.Payload = payload
+	// make a copy so the proto initialization doesn't modify the original entry
+	overheadClone := proto.Clone(logOverhead)
+	overheadBytes := proto.Size(overheadClone)
+
+	payload, splits, err := parseEntryPayload(log.Body(), l.maxEntrySize-overheadBytes)
+	if err != nil {
+		return []logging.Entry{entry}, logName, err
+	}
+
+	// Split log entries with a string payload into fewer entries
+	if splits > 1 {
+		entries := make([]logging.Entry, int(splits))
+		payloadString := payload.(string)
+
+		// Start by assuming all splits will be even (this may not be the case)
+		startIndex := 0
+		endIndex := int(math.Floor((1.0 / splits) * float64(len(payloadString))))
+		for i := 1; i <= int(splits); i++ {
+			currentSplit := payloadString[startIndex:endIndex]
+
+			// If the current split is larger than the entry size, iterate until it is within the max
+			// (This may happen since not all characters are exactly 1 byte)
+			for len([]byte(currentSplit)) > l.maxEntrySize {
+				endIndex--
+				currentSplit = payloadString[startIndex:endIndex]
+			}
+			entries[i-1] = entry
+			entries[i-1].Payload = currentSplit
 
-	return entry, nil
+			// Update slice indices to the next chunk
+			startIndex = endIndex
+			endIndex = int(math.Floor((float64(i+1) / splits) * float64(len(payloadString))))
+		}
+		return entries, logName, nil
+	}
+
+	entry.Payload = payload
+	return []logging.Entry{entry}, logName, nil
 }
 
-func parseEntryPayload(logBody pcommon.Value) (interface{}, error) {
+func parseEntryPayload(logBody pcommon.Value, maxEntrySize int) (interface{}, float64, error) {
```

Comment:
Should we return an int instead of a float for the number of splits?

Reply:
yeah, it's the same number of casts either way but it's a little easier to read as an int